partner_communication_switzerland/models/contracts.py

```diff
@@ -145,6 +152,9 @@ def send_communication(self, communication, correspondent=True,
                               (overrides the previous parameter)
         :return: communication created recordset
         """
+        if not self.send_introduction_letter and \
```

Comment:
This test is wrong and should not be placed here. Instead, put it here, where the letter is processed: https://github.com/CompassionCH/compassion-switzerland/blob/26d37a551884a488e8eddcd34d57b89ef62b6d48/sbc_switzerland/models/correspondence.py#L126
The test would more be like:
```
intro_letter = self.env.ref('sbc_compassion.correspondence_type_new_sponsor')
if intro_letter in letter.communication_type_ids and not letter.sponsorship_id.send_introduction_letter:
    continue
```